tests/phpunit/tests/url.php

```diff
@@ -569,4 +569,69 @@ public function test_url_functions_for_dots_in_paths() {
 			);
 		}
 	}
+
+	/**
+	 * Test get_adjacent_post with posts having identical post_date.
+	 *
+	 * @ticket 8107
+	 * @covers ::get_adjacent_post
+	 */
+	public function test_get_adjacent_post_with_identical_dates() {
+		$identical_date = gmdate( 'Y-m-d H:i:s', time() );
+
+		// Create 3 posts with identical dates but different IDs.
+		$post_ids = array();
+		for ( $i = 1; $i <= 3; $i++ ) {
+			$post_ids[] = self::factory()->post->create(
+				array(
+					'post_title' => "Identical Post $i",
+					'post_date'  => $identical_date,
+				)
+			);
+		}
+
+		if ( ! isset( $GLOBALS['post'] ) ) {
+			$GLOBALS['post'] = null;
+		}
+		$orig_post = $GLOBALS['post'];
```

Comment:
This is not needed since the globals get reset with each test:

https://github.com/WordPress/wordpress-develop/blob/aae3ac32369a78cbf40472aa8cd39a74481cf687/tests/phpunit/includes/abstract-testcase.php#L173-L177

```suggestion
```